<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConfigureEquationParametersWindow</class>
 <widget class="QWidget" name="ConfigureEquationParametersWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>616</width>
    <height>596</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Configure equation parameters</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources/apprcc.qrc">
    <normaloff>:/APThrehold_icon.png</normaloff>:/APThrehold_icon.png</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">
QGroupBox {
    border: 2px solid #829aa0; /* 设置2px的边框颜色 */
    border-radius: 5px;        /* 设置圆角 */
    margin-top: 10px;          /* 标题与内容的距离 */
}

QGroupBox::title {
    subcontrol-origin: margin;  /* 标题的来源 */
    subcontrol-position: top left; /* 标题位置 */
    padding: 0 3px;  /* 为标题添加左右内边距 */
 
    color: black;             /* 标题的文字颜色 */
    font-weight: bold;          /* 标题的加粗 */
}

</string>
  </property>
  <layout class="QGridLayout" name="gridLayout" rowstretch="0,0,0,0">
   <property name="leftMargin">
    <number>9</number>
   </property>
   <property name="topMargin">
    <number>9</number>
   </property>
   <property name="rightMargin">
    <number>9</number>
   </property>
   <property name="bottomMargin">
    <number>9</number>
   </property>
   <property name="verticalSpacing">
    <number>6</number>
   </property>
   <item row="3" column="0">
    <widget class="QPushButton" name="pushButton_OK">
     <property name="text">
      <string>OK</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QPushButton" name="pushButton_canel">
     <property name="text">
      <string>Canel</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Optimization</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Algorithm</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QPlainTextEdit" name="plainTextEdit_optimization">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="comboBox_algorithm">
        <item>
         <property name="text">
          <string>CMA-ES</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="pushButton_run">
        <property name="text">
         <string>Run</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="pushButton_terminate">
        <property name="text">
         <string>Terminate</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Objective function</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="0" column="1">
       <widget class="QComboBox" name="comboBox_objective_function">
        <item>
         <property name="text">
          <string>Gama factor</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Option</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QStackedWidget" name="stackedWidget_objective_function">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="page_3">
         <layout class="QGridLayout" name="gridLayout_6">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_time_window_gama_factor">
            <property name="value">
             <double>0.840000000000000</double>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>Time window σ (ms)</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_4"/>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Equation parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QStackedWidget" name="stackedWidget_equation_parameters">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="page">
         <layout class="QGridLayout" name="gridLayout_4">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="5" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_Vi_dynamics_min">
            <property name="minimum">
             <double>-200.000000000000000</double>
            </property>
            <property name="maximum">
             <double>0.000000000000000</double>
            </property>
            <property name="value">
             <double>-90.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="4" column="2">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_ki_dynamics_max">
            <property name="minimum">
             <double>-100.000000000000000</double>
            </property>
            <property name="maximum">
             <double>0.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_Vi_dynamics">
            <property name="text">
             <string>Vi</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_ki_dynamics_min">
            <property name="minimum">
             <double>-100.000000000000000</double>
            </property>
            <property name="maximum">
             <double>0.000000000000000</double>
            </property>
            <property name="value">
             <double>-20.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_VT_dynamics">
            <property name="text">
             <string>VT</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_alpha_dynamics_min"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_alpha_dynamics">
            <property name="text">
             <string>α</string>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_ka_dynamics_max">
            <property name="value">
             <double>20.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_ka_dynamics">
            <property name="text">
             <string>ka</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_tao_theta_dynamics_min">
            <property name="minimum">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.100000000000000</double>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_ki_dyamics">
            <property name="text">
             <string>ki</string>
            </property>
           </widget>
          </item>
          <item row="5" column="4">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_Vi_dynamics_fixed">
            <property name="minimum">
             <double>-2000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>9900.989999999999782</double>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_tao_theta_dynamics">
            <property name="text">
             <string>τθ</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string>Name</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="3">
           <widget class="QLabel" name="label_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string>Bounds</string>
            </property>
           </widget>
          </item>
          <item row="2" column="4">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_alpha_dynamics_fixed">
            <property name="minimum">
             <double>-1000000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1000000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="0" column="4" colspan="2">
           <widget class="QLabel" name="label_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string>Fixed value</string>
            </property>
           </widget>
          </item>
          <item row="4" column="4">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_ki_dynamics_fixed">
            <property name="minimum">
             <double>-2000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>2000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="6" column="4">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_VT_dynamics_fixed">
            <property name="minimum">
             <double>-20000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1000000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_alpha_dynamics_max">
            <property name="value">
             <double>1.570000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_tao_theta_dynamics_max">
            <property name="value">
             <double>10.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="5" column="2">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_Vi_dynamics_max"/>
          </item>
          <item row="3" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_ka_dynamics_min"/>
          </item>
          <item row="6" column="2">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_VT_dynamics_max"/>
          </item>
          <item row="3" column="4">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_ka_dynamics_fixed">
            <property name="minimum">
             <double>-10100.000000000000000</double>
            </property>
            <property name="maximum">
             <double>100000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="4">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_tao_theta_dynamics_fixed">
            <property name="minimum">
             <double>-1000000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1000000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_VT_dynamics_min">
            <property name="minimum">
             <double>-200.000000000000000</double>
            </property>
            <property name="maximum">
             <double>100.000000000000000</double>
            </property>
            <property name="value">
             <double>-90.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="3" rowspan="6">
           <widget class="QCheckBox" name="checkBox_fit_dynamics">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>20</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="iconSize">
             <size>
              <width>16</width>
              <height>14</height>
             </size>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="5" rowspan="6">
           <widget class="QCheckBox" name="checkBox_fixed_dynamics">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>20</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_2"/>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../resources/apprcc.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>stackedWidget_equation_parameters</sender>
   <signal>currentChanged(int)</signal>
   <receiver>stackedWidget_equation_parameters</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>307</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>112</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
